jupyterhub/singleuser.py

```diff
@@ -233,13 +241,13 @@ def _validate_base_url(self, proposal):
     def _port_default(self):
         if os.environ.get('JUPYTERHUB_SERVICE_URL'):
             url = urlparse(os.environ['JUPYTERHUB_SERVICE_URL'])
-            return url.port
+            return url.port or 8888
```

Comment:
@minrk Do you want this inside the `if` block or outside of it? (Same question for line 250.)